runtime/drivers/azure/azure.go

```diff
@@ -0,0 +1,332 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
+	"github.com/bmatcuk/doublestar/v4"
+	"github.com/mitchellh/mapstructure"
+	"github.com/rilldata/rill/runtime/drivers"
+	rillblob "github.com/rilldata/rill/runtime/drivers/blob"
+	"github.com/rilldata/rill/runtime/pkg/activity"
+	"github.com/rilldata/rill/runtime/pkg/globutil"
+	"go.uber.org/zap"
+	"gocloud.dev/blob"
+	"gocloud.dev/blob/azureblob"
+)
+
+func init() {
+	drivers.Register("azure", driver{})
+	drivers.RegisterAsConnector("azure", driver{})
+}
+
+var spec = drivers.Spec{
+	DisplayName:        "Azure Blob Storage",
+	Description:        "Connect to Azure Blob Storage.",
+	ServiceAccountDocs: "https://docs.rilldata.com/deploy/credentials/azure",
+	SourceProperties: []drivers.PropertySchema{
+		{
+			Key:         "path",
+			DisplayName: "Blob URI",
+			Description: "Path to file on the disk.",
+			Placeholder: "azblob://container-name/path/to/file.csv",
+			Type:        drivers.StringPropertyType,
+			Required:    true,
+			Hint:        "Glob patterns are supported",
+		},
+		{
+			Key:         "azure.storage.account",
+			DisplayName: "Azure Storage Account",
+			Description: "Azure Storage Account inferred from your local environment.",
+			Type:        drivers.InformationalPropertyType,
+			Hint:        "Set your local credentials: <code>az login</code> Click to learn more.",
+			Href:        "https://docs.rilldata.com/develop/import-data#configure-credentials-for-azure",
+		},
+	},
+	ConfigProperties: []drivers.PropertySchema{
+		{
+			Key:    "azure_storage_account",
+			Secret: true,
+		},
+		{
+			Key:    "azure_storage_key",
+			Secret: true,
+		},
+		{
+			Key:    "azure_storage_sas_token",
+			Secret: true,
+		},
+	},
+}
+
+type driver struct{}
+
+type configProperties struct {
+	Account         string `mapstructure:"azure_storage_account"`
+	Key             string `mapstructure:"azure_storage_key"`
+	SASToken        string `mapstructure:"azure_storage_sas_token"`
+	StorageDomain   string `mapstructure:"azure_storage_domain,default:blob.core.windows.net"`
+	IsCDN           bool   `mapstructure:"azure_storage_is_cdn,default:false"`
+	IsLocalEmulator bool   `mapstructure:"azure_storage_is_local_emulator,default:false"`
```

Comment:
As I understood it is only for developmental purposes so we can ignore `azure_storage_is_local_emulator`.
`azure_storage_domain` and `azure_storage_is_cdn` seems to be optional configs. Let's ignore them and only add if some user asks for them otherwise we need to take the values every time we do `rill env configure`.